### PR TITLE
Use scrypt with random salt for key derivation

### PR DESCRIPTION
**Pull request by Mrtenz**

### CODE REVIEW

**Comment by Mrtenz** [comment]
[sc-8978]

File: packages/signer/src/utils/encryption.spec.ts [hunk 1]
```diff
@@ -31,7 +32,7 @@ describe('createEncryptionKey', () => {
 
 describe('hashPassword', () => {
   it('correctly SHA256 hashes a string', async () => {
```
Comment (FrederikBolding) [inline packages/signer/src/utils/encryption.spec.ts:34]: Update this to be correct

File: packages/signer/src/utils/encryption.ts [hunk 2]
```diff
@@ -52,10 +46,20 @@ export const decrypt = (data: string, key: Buffer) => {
...
-  return pbkdf2(password, SALT, ITERATIONS, KEY_LENGTH, HASH_ALGORITHM);
+  // `util.promisify` doesn't properly infer the types of Scrypt, so it's wrapped in a promise here
+  // instead
+  return new Promise((resolve, reject) => {
+    crypto.scrypt(password, salt, KEY_LENGTH, (error, key) => {
+      if (error) {
+        return reject(error);
+      }
+
+      resolve(key);
```
Comment (FrederikBolding) [inline packages/signer/src/utils/encryption.ts:62]: Missing return?

File: packages/signer/src/crypto/secrets.test.ts [hunk 3]
```diff
@@ -127,10 +136,14 @@ describe('hasSettingsKey', () => {
 
 describe('checkSettingsKey', () => {
   it('checks if the settings key can be decrypted', async () => {
-    (keytar.getPassword as jest.MockedFunction<typeof keytar.getPassword>).mockImplementation(
-      async () =>
-        '6a7cd51df01e5152968a5acbd312c2d55c08eb7cf4061c1a4d1de326ebcbbea026d35cd7b5c43ade2d21938ada7a165c39c8f3fd'
-    );
+    (keytar.getPassword as jest.MockedFunction<typeof keytar.getPassword>)
+      .mockImplementationOnce(
+        async () => 'dc509bcfc343f1bebb4d75749695fd3ef204306f07eff6f86eec91587ba03bbc'
```
Comment (FrederikBolding) [inline packages/signer/src/crypto/secrets.test.ts:141]: Can you define this globally if it is re-used for all tests?

File: packages/signer/src/crypto/secrets.ts [hunk 4]
```diff
@@ -19,7 +19,7 @@ export const clearEncryptionKey = () => {
 };
 
 export const init = async (password: string) => {
-  const key = await hashPassword(password);
+  const key = await hashPassword(password, await getSalt());
```
Comment (FrederikBolding) [inline packages/signer/src/crypto/secrets.ts:22]: Maybe add a const for the salt for readability?

**Review by FrederikBolding — CHANGES_REQUESTED** [review]

**Review by FrederikBolding — APPROVED** [review]
lgtm - ask cure53 if we should be tweaking scrypt params

**Pull request merged**